### PR TITLE
materials for the simulation of dirts BNB events added

### PR DESCRIPTION
**Pull request by cfarnese**
I added here few things that can be used to produce dirt events for the BNB beam! The code I must confess is not so "beautiful" but I commit it in any case to try to have it for the new release! In particular there are :

- a couple of fhicl files in fcl/gen/genie to generate the dirts events

- a filter module and its fhicl to remove the neutrino interactions with the vertex inside the active volume or that did not deposit energy in the detector

### CODE REVIEW

**Comment by SFBayLaser** [comment]
trigger build

**Comment by SFBayLaser** [comment]
Interesting question... should we require "beauty" before we sign off on a PR?

**Comment by cfarnese** [comment]
I believe usually the code should be clear and as simple as possible... I must confess in this case I prefer to make something "quick and dirt" because the main goal was to arrive in time and unfortunately the many activities don't provide me the proper time to make the code clearer.. sorry for this!

**Comment by mmrosenberg** [comment]
trigger build

**Review by SFBayLaser — APPROVED** [review]
Ok, quick and dirty… but approving!

File: icaruscode/Filters/FilterDirts_module.cc
Comment (PetrilloAtWork) [inline icaruscode/Filters/FilterDirts_module.cc:28]: Unused?
```suggestion
```

File: icaruscode/Filters/FilterDirts_module.cc
Comment (PetrilloAtWork) [inline icaruscode/Filters/FilterDirts_module.cc:18]: Unused?
```suggestion
```

**Review by PetrilloAtWork — CHANGES_REQUESTED** [review]
The name of `SimChannel` data product needs to be moved into a configuration parameter.
The rest are suggestions.

**Review by PetrilloAtWork — APPROVED** [review]
The comments from my previous review persist.
The submitter has explained me the role of `GenInTimeSorter`, which indeed hides a bug.
The sorter will take all generated stuff, cosmic rays from CORSIKA and neutrino interactions from GENIE, and will split them in "in time" and "out of time". The first category is then assumed to have been GEANT'd already, while the second category will be processed through GEANT4 in the next stage (e.g. `intime_g4_icarus.fcl`) and merged with the GEANT4 output of this stage.
The problem is that in-time cosmic rays, if any, will be assumed to have been processed already, while that's not the case (here GEANT4 only processed the neutrino events), resulting in the neutrino events never having a cosmic ray close in time.
The probability of such overlap is not large, though (10-kHz cosmic rays appear in a 2-µs gate once every 50 times).

With the submitter we agreed that the points raised in my review will be addressed ideally next week, and anyway in time for the production release. In the meanwhile, this PR should be merged as is in time for the _"intermediate"_ validation release.

**To the release managers: please do not delete the branch after merging.**

**Comment by cfarnese** [comment]
I Introduce with the last commits almost all the corrections suggested by Gianluca! I tested this morning the code and it provides compatible results with respect to the previous version!

**Comment by mmrosenberg** [comment]
trigger build

**Comment by mmrosenberg** [comment]
trigger build

**Pull request merged**
